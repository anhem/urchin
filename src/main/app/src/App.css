#urchin {
    display: table;
    width: 100vh;
    height: 100vh;
}