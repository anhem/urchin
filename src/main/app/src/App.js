import React, {Component} from 'react';
import {Route, Router, Switch} from 'react-router-dom'
import history from './history'
import {NotificationContainer} from 'react-notifications';
import Root from './component/Root';
import LoginContainer from './component/LoginContainer';
import ScrollToTop from "./component/ScrollToTop";

class App extends Component {
    render() {
        return (
            <div className="urchin">
                <NotificationContainer/>
                <Router history={history}>
                    <ScrollToTop>
                        <Switch>
                            <Route path='/login'
                                   component={LoginContainer}/>
                            <Route path='/'
                                   component={Root}/>
                        </Switch>
                    </ScrollToTop>
                </Router>
            </div>
        );
    }
}

export default App;
