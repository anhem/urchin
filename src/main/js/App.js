import React, {Component} from "react";
import rest from "rest";
import Groups from "./Groups";

class App extends Component {

    constructor(props) {
        super(props);
        this.state = {groups: []};
    }

    componentDidMount() {
        rest({
            method: 'GET',
            path: '/api/groups'
        }).then(response => {
            this.setState({groups: JSON.parse(response.entity).data});
        });
    }

    render() {
        return (
            <Groups groups={this.state.groups}/>
        );
    }
}

export default App;