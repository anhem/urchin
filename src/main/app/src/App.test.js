import React from 'react';
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import App from './App';

describe('App', () => {

    it('match snapshot', () => {
        expect(toJson(shallow(<App/>))).toMatchSnapshot();
    });
});
