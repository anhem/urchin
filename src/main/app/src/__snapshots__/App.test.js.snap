// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App match snapshot 1`] = `
<div
  className="urchin"
>
  <NotificationContainer
    enterTimeout={400}
    leaveTimeout={400}
  />
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(ScrollToTop)>
      <Switch>
        <Route
          component={[Function]}
          path="/login"
        />
        <Route
          component={[Function]}
          path="/"
        />
      </Switch>
    </withRouter(ScrollToTop)>
  </Router>
</div>
`;
